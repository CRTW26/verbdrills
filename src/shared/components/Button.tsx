import React from 'react'

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  text: string
}

const Button: React.FC<Props> = ({ text, ...rest }) => {
  return (
    <>
      <button {...rest}>{text}</button>

      <style jsx>{`
        .btn {
          background: none;
          border: none;
          border-radius: 5px;
          padding: 0.5rem 1rem;
          width: 100%;
        }

        .btn--primary {
          background: #e0ff4f;
          color: #0b3954;
        }
      `}</style>
    </>
  )
}

export default Button
