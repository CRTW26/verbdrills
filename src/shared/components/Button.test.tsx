import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Button from './Button'

describe('<Button />', () => {
  test('should render', () => {
    const { asFragment } = render(<Button text="Practice" />)

    expect(asFragment()).toMatchSnapshot()
  })
})
