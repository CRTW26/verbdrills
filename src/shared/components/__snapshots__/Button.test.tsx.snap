// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render 1`] = `
<DocumentFragment>
  <button
    class="jsx-3131124250 "
  >
    Practice
  </button>
</DocumentFragment>
`;
